src/main/java/com/nooblol/account/service/impl/MatchGameInfoServiceImpl.java

```diff
@@ -51,13 +53,61 @@ public class MatchGameInfoServiceImpl implements MatchGameInfoService {
   private final HttpHeaders initRiotHeader;
 
   private final MatchGameInfoMapper matchGameInfoMapper;
+  private final MatchGameAddInfoMapper matchGameAddInfoMapper;
 
+  /**
+   * DB의 게임 전적에 대하여 바로 Return 을 진행 하나, 데이터가 존재하지 않는 경우에는 동기화를 한 이후에 조회를 진행함)
+   */
+
+  @Override
+  public ResponseDto getMatchInfoListByPuuid(String puuid, int pageNum) throws Exception {
+    if (StringUtils.isBlank(puuid)) {
+      throw new IllegalArgumentException("PuuId가 입력되지 않았습니다.");
+    }
+    List<MatchGameSimpleDto> matchResult = selectMatchSimpleListByPuuidInDB(puuid, pageNum);
+
+    if (!ObjectUtils.isEmpty(matchResult)) {
+      return new ResponseDto(HttpStatus.OK.value(), matchResult);
+    }
+
+    return syncRiotToDbByPuuidAfterGetMatchSimpleList(puuid, pageNum);
+  }
+
+  @Override
+  @Transactional(readOnly = true)
+  public List<MatchGameSimpleDto> selectMatchSimpleListByPuuidInDB(String puuid, int pageNum) {
+    Map<String, Object> searchParam = new HashMap<>();
+    searchParam.put("puuid", puuid);
+    searchParam.put("pageNum", pageNum);
+
+    List<MatchGameSimpleDto> selectMatchSimpleList =
+        matchGameAddInfoMapper.selectMatchSimpleList(searchParam);
+
+    if (!ObjectUtils.isEmpty(selectMatchSimpleList)) {
+      selectMatchSimpleList.stream().forEach(matchGameSimpleDto -> {
+        String matchId = matchGameSimpleDto.getMatchId();
+        List<MatchGameSimpleDto> participantsList =
+            matchGameAddInfoMapper.selectMatchSimpleParticipantsList(matchId);
+        matchGameSimpleDto.setParticipants(participantsList);
+      });
+    }
+    return selectMatchSimpleList;
```

Comment:
해당 부분에 대해서는 PR작성후 지금 다시 보게 되었는데 아래 코드로 수정하고자 합니다 ㅎㅎ....
```suggestion
    if (ObjectUtils.isEmpty(selectMatchSimpleList)) {
      return selectMatchSimpleList;
    }
    selectMatchSimpleList.stream().forEach(
        matchGameSimpleDto -> {
            String matchId = matchGameSimpleDto.getMatchId();
            List<MatchGameSimpleDto> participantsList =
                matchGameAddInfoMapper.selectMatchSimpleParticipantsList(matchId);
            matchGameSimpleDto.setParticipants(participantsList);
    });
```

Reply:
👍